diff --git a/libcore/pixmapcontainer.c b/libcore/pixmapcontainer.c
index 69a47ca..99cb743 100644
--- a/libcore/pixmapcontainer.c
+++ b/libcore/pixmapcontainer.c
@@ -318,7 +318,7 @@ void cPixmapContainer::FadeOut(void) {
         }
         DoFlush();
         int Delta = cTimeMs::Now() - Now;
-        if (Running() && (Delta < FadeFrameTime))
+        if (Delta < FadeFrameTime)
             cCondWait::SleepMs(FadeFrameTime - Delta);
         if ((int)(Now - Start) > fadeTime)
             break;
