diff --git a/p8-platform-config.cmake.in b/p8-platform-config.cmake.in
index 47f30f1..e48188b 100644
--- a/p8-platform-config.cmake.in
+++ b/p8-platform-config.cmake.in
@@ -19,7 +19,7 @@ set (p8-platform_CONFIG_VARS "@p8-platform_CONFIG_VARS@")
 if(WIN32)
   set (p8-platform_LIBRARY "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/p8-platform.lib")
 else(WIN32)
-  set (p8-platform_LIBRARY "-L@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@ -lp8-platform")
+  set (p8-platform_LIBRARY "-lp8-platform")
 endif(WIN32)
 set (p8-platform_LIBRARIES ${p8-platform_LIBRARY} "@p8-platform_LIBRARIES@")
 mark_as_advanced (p8-platform_LIBRARY)
