--- a/enigma.c
+++ b/enigma.c
@@ -1380,7 +1380,7 @@
   }

   lineHeight = pFontList->Height();
-  nMarkerGap = min(MarkerGap, lineHeight / 2 - 1); //lineHeight - 2 * MarkerGap
+  nMarkerGap = std::min(MarkerGap, lineHeight / 2 - 1); //lineHeight - 2 * MarkerGap
   xItemLeft = xBodyLeft + (EnigmaConfig.showMarker ? lineHeight : ListHBorder);
   xItemRight = (fShowLogo || fShowInfo ? xBodyRight : xInfoRight) - ListHBorder - SmallGap - SmallGap - SmallGap;
   int numItems = MaxItems();
@@ -3808,7 +3808,7 @@
   }

   lineHeight = pFontListItem->Height();
-  nMarkerGap = min(MarkerGap, lineHeight / 2 - 1); //lineHeight - 2 * MarkerGap
+  nMarkerGap = std::min(MarkerGap, lineHeight / 2 - 1); //lineHeight - 2 * MarkerGap
   int LogoSize = IconHeight;
   LogoSize += (LogoSize % 2 ? 1 : 0);
   currentIndex = -1;